python/paddle/fluid/transpiler/distribute_transpiler.py

```diff
@@ -327,10 +336,11 @@ def transpile(self,
                 })
 
         if self.sync_mode:
+            # form a WAW dependency
             program.global_block().append_op(
                 type="fetch_barrier",
                 inputs={},
```

Comment:
If so, recv_op must add that output dummy variable, since recv_op use output variable list to retrieve from the parameter server, so there must be a runtime check, which may harm performance.

Let fetch_barrier output the same list as recv_op, ssa graph builder will automatically add a WAS deps var, this can do the same thing.